dashboard/src/components/api/npm-registry.factory.ts

```diff
@@ -0,0 +1,65 @@
+/*
```

Comment:
I would assume the managing of plugins would be server side instead of client side (like searching all extensions).
So it doesn't use client connection to download plugins but use server side.

But it may be for step 2